### DISCUSSION TITLE
Question: Is it possible to use DKG algorithm in smart contract?

### DISCUSSION THREAD

**Discussion by StackOverflowExcept1on** [ZcashFoundation/frost#732] [Q&A] [answered]
https://github.com/ZcashFoundation/frost/blob/dcf17732f791cd5c69aeed5bb4ff60d019a57ee8/frost-core/src/keys/dkg.rs#L18-L23
https://frost.zfnd.org/terminology.html#broadcast-channel
Can smart contract act as "secure broadcast channel"?

**Comment by conradoplg** [comment, marked as answer]
It depends on a lot of factors (How participants check which published commitments belong to them? How exactly the smart contract works?) but in general, yes, since it could provide an immutable public record of the commitments.

**Comment by pool2win** [comment]
How can we handle reorgs, if the chain that has smart contracts supports reorgs?

We just wait for enough confirmations?

**Reply by conradoplg** [reply]
Yes, wait for enough confirmations (It's not a particularly efficient solution.)

**Reply by StackOverflowExcept1on** [reply]
It depends on the blockchain, for example, there are blockchains with delayed self-execution of smart contracts.
